### ISSUE TITLE
CollectionDB is wiped every time device is searched for songs

### ISSUE THREAD

**Issue by sagarjs** [tomahawk-player/tomahawk-android#25]
The collection DB is wiped every time the device folders are processed for tracks. Is there a specific reason for this? Is it possible to update the table instead of wiping and recreating every time?

The wipe is happening in the function "processMediaWrappers" in the file org.tomahawk.libtomahawk.collection.UserCollection.

This function calls the wipe() function and processes tracks to create UserCollection data.

The function processMediaWrappers is called in run() function in the class GetMediaItemsRunnable.

So it seems like, while tracks are added and removed on every search, the UserCollection is wiped and recreated every time.

**Comment by mrmaffen** [comment]
Yeah, the current implementation is definitely not the best and most clean, but it works.
The majority of the work in terms of searching for local tracks is actually being done before processMediaWrappers is being called. In the GetMediaItemsRunnable all tracks are being searched for and parsed. Then the resulting MediaWrapper objects are stored in the database "DatabaseHelper.get().addMedias(mediaWrappers);". This MediaWrapper database is not being wiped and re-filled every time.
All that's done in processMediaWrappers is to iterate through all MediaWrappers and parse them into ScriptResolverTrack objects. Those objects are then being stored in another database (UserCollectionDb), which is the one you were talking about. It's being wiped and re-filled every time. 
This implementation with 2 separate databases is mostly just a remnant from a previous implementation.

A better way to do it would of course be to get rid of the UserCollectionDb entirely and pull all the data directly from the MediaWrapper database.

Please always feel free to post PRs for these kinds of things.
